sycl/include/CL/sycl/handler.hpp

```diff
@@ -721,14 +725,24 @@ class __SYCL_EXPORT handler {
   // NOTE: the name of this function - "kernel_single_task" - is used by the
   // Front End to determine kernel invocation kind.
   template <typename KernelName, typename KernelType>
-  __attribute__((sycl_kernel)) void kernel_single_task(KernelType KernelFunc) {
+  __attribute__((sycl_kernel)) void
+#ifdef __SYCL_NONCONST_FUNCTOR__
+  kernel_single_task(KernelType KernelFunc) {
+#else
+  kernel_single_task(const KernelType &KernelFunc) {
+#endif
```

Comment:
```
#if !CL_SYCL_LANGUAGE_VERSION || CL_SYCL_LANGUAGE_VERSION < 2020
#define __SYCL_CONST_FUNCTOR__ const
#else 
#define __SYCL_CONST_FUNCTOR__
#endif
```


```suggestion
  kernel_single_task(__SYCL_CONST_FUNCTOR__ KernelType KernelFunc) {
```

Reply:
Else branch takes reference type parameter.
Probably we can re-use `KERNELFUNCPARAM` macro here, but I suggest re-name it to start with `_` to avoid conflicts with user defined macro definitions.

Reply:
renamed to _KERNELFUNCPARAM and reused